integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+                        SecretBundleConfigSource.this.mostDistantExpirationInstant; // volatile read
+                    for (SecretSummary ss : secretSummaries) {
+                        tasks.add(() -> {
+                                valueNodes.put(ss.getSecretName(), valueNode(secrets, ss, decoder));
+                                return null;
+                            });
+                        java.util.Date d = ss.getTimeOfCurrentVersionExpiry();
+                        Instant i = d == null ? null : d.toInstant();
+                        if (i != null && (mostDistantExpirationInstant == null || mostDistantExpirationInstant.isBefore(i))) {
+                            mostDistantExpirationInstant = i;
+                        }
+                    }
+                    SecretBundleConfigSource.this.mostDistantExpirationInstant = mostDistantExpirationInstant; // volatile write
+                    completeAllSecretsTasks(tasks, secrets);
+                    ObjectNode.Builder onb = ObjectNode.builder();
+                    for (Entry<String, ValueNode> e : valueNodes.entrySet()) {
+                        onb.addValue(e.getKey(), e.getValue());
+                    }
+                    return Optional.of(NodeContent.builder()
+                                       .node(onb.build())
+                                       .build());
+                };
+            }
+        }
+
+
+        /*
+         * Instance methods.
+         */
+
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // PollableSource
+        public boolean isModified(Instant instant) {
+            Instant i = this.mostDistantExpirationInstant;
+            return i == null || i.isBefore(instant == null ? now() : instant);
+        }
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // NodeConfigSource
+        public Optional<NodeContent> load() {
+            return this.loader.get();
+        }
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // PollableSource
+        public Optional<PollingStrategy> pollingStrategy() {
+            return super.pollingStrategy();
+        }
+
+        private Optional<NodeContent> absentNodeContent() {
+            this.mostDistantExpirationInstant = null; // volatile write
+            return ABSENT_NODE_CONTENT;
+        }
+
+
+        /*
+         * Static methods.
+         */
+
+
+        private static Builder builder() {
+            return new Builder();
+        }
+
+        private static void completeAllSecretsTasks(Collection<? extends Callable<Void>> tasks, AutoCloseable secrets) {
+            RuntimeException re = null;
+            try (ExecutorService es = newVirtualThreadPerTaskExecutor()) {
+                for (Future<?> future : es.invokeAll(tasks)) {
+                    try {
+                        future.get();
+                    } catch (RuntimeException e) {
```

Comment:
considering the duplication here at a few levels, wouldn't it be easier to just write a `handleException(AtomicReference<RuntimeException> re, Exception e)` method and just always delegate to it at everywhere to handle a plain old catch (Exception e)?

Reply:
I will look at exception handling again. In some cases I need to handle the fact that a client is `AutoCloseable`, but does not override `close()`, so it throws `Exception`, and in some of those cases there are multiple exceptions that can be thrown from within the try/finally block, and I want them all captured. In those cases, frequently I need to suppress other exceptions. It's complicated, and I agree that if there is a way to reduce the complexity I should take it.

Reply:
Done